src/rust/cryptography-x509/src/extensions.rs

```diff
@@ -73,6 +72,12 @@ pub struct Extension<'a> {
     pub extn_value: &'a [u8],
 }
 
+impl<'a> Extension<'a> {
+    pub fn value<T: asn1::Asn1Readable<'a>>(&'a self) -> asn1::ParseResult<T> {
+        asn1::parse_single(self.extn_value)
+    }
+}
```

Comment:
This probably makes this PR a fair bit larger, but I think we have a bunch of use cases for this in cryptography already, notably: `src/rust/src/x509/common.rs`, line 412. we should probably change that to pass an Extension value to the callback, and then use `value()` there.

Reply:
[f4a5664](https://github.com/pyca/cryptography/pull/9213/commits/f4a56648da2318db9f85564df05179bff0816ab7) should cover this.